project_budget_management/models/project_project.py

```diff
@@ -0,0 +1,183 @@
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from datetime import timedelta
+
+from odoo import api, fields, models
+
+
+class ProjectProjectStage(models.Model):
+    _inherit = 'project.project.stage'
+
+    is_close = fields.Boolean("Closing Stage")
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+
+    def _compute_spent_budget(self):
+        """
+        Compute the Spent Budget of Project
+        """
+        timesheet_invoice_obj = self.env["timesheet.invoice"]
+        for project in self:
+            timesheet_invoice_records = timesheet_invoice_obj.sudo().search(
+                [
+                    ("project_id", "=", project.id),
+                    ("state", "in", ("approved", "completed")),
+                ]
+            )
+            project.spent_budget = sum(
+                inv.total_amount for inv in timesheet_invoice_records
+            )
+
+    def _compute_running_cost(self):
+        """
+        Compute Running Cost of Project.
+        """
+        timesheet_invoice_obj = self.env["timesheet.invoice"]
+        for project in self:
+            timesheet_invoice_records = timesheet_invoice_obj.search(
+                [
+                    ("project_id", "=", project.id),
+                    ("state", "in", ("draft", "confirm", "pre-approved")),
+                ]
+            )
+            project.running_cost = sum(
+                inv.total_amount for inv in timesheet_invoice_records
+            )
+
+    @api.depends("actual_budget", "revised_budget")
+    def _compute_actual_revised_budget(self):
+        for project in self:
+            actual_revised_budget = project.actual_budget
+            if project.revised_budget > 0:
+                actual_revised_budget = project.revised_budget
+            project.actual_revised_budget = actual_revised_budget
```

Comment:
Simplify this as 
project.actual_revised_budget = project.revised_budget if project.revised_budget > 0 else project.actual_budget